webui/react/src/pages/FlatRuns/FlatRuns.tsx

```diff
@@ -515,22 +516,18 @@ const FlatRuns: React.FC<Props> = ({ projectId, workspaceId, searchId }) => {
     if (isLoadingSettings || Loadable.isNotLoaded(loadableFormset)) return;
     try {
       const filters = JSON.parse(filtersString);
+      const filterFormSet = JSON.parse(filtersString);
       if (searchId) {
         // only display trials for search
-        const existingFilterGroup = { ...filters.filterGroup };
         const searchFilter = {
           columnName: 'experimentId',
-          kind: 'field',
-          location: 'LOCATION_TYPE_RUN',
-          operator: '=',
-          type: 'COLUMN_TYPE_NUMBER',
+          kind: 'field' as const,
+          location: V1LocationType.RUN,
+          operator: Operator.Eq,
+          type: V1ColumnType.NUMBER,
           value: searchId,
         };
-        filters.filterGroup = {
-          children: [existingFilterGroup, searchFilter],
-          conjunction: 'and',
-          kind: 'group',
-        };
+        combine(filterFormSet.filterGroup, 'and', searchFilter);
```

Comment:
`combine` doesn't mutate the filter form set, so you'll still need to set the result of this to the variable.